File: Backend/DigitalStore.Domain/Entities/Comment.cs
using System;

namespace DigitalStore.Domain.Entities
{
    public class Comment
    {
        public long Id { get; set; }
        public int UserId { get; set; }
        public int TargetId { get; set; }
        public byte TargetType { get; set; } // 1=Question, 2=Answer, 3=EducationContent
        public long? ParentCommentId { get; set; }
        public string Content { get; set; } = string.Empty;
        public int LikeCount { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime CreatedAt { get; set; }

        // Extra properties mapped from SQL Join
        public string UserDisplayName { get; set; } = string.Empty;
        public string? UserAvatar { get; set; }
        public bool IsLikedByCurrentUser { get; set; }
    }
}

File: education_app/lib/features/comment/domain/entities/comment.dart
import 'package:equatable/equatable.dart';

class Comment extends Equatable {
  final int id;
  final int userId;
  final String userDisplayName;
  final String? userAvatar;
  final String content;
  final int likeCount;
  final String createdAt;
  final List<Comment> replies;
  final bool isLikedByCurrentUser;

  const Comment({
    required this.id,
    required this.userId,
    required this.userDisplayName,
    this.userAvatar,
    required this.content,
    required this.likeCount,
    required this.createdAt,
    this.replies = const [],
    this.isLikedByCurrentUser = false,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        userDisplayName,
        userAvatar,
        content,
        likeCount,
        createdAt,
        replies,
        isLikedByCurrentUser,
      ];
}
